core/src/main/java/org/apache/accumulo/core/file/rfile/RelativeKey.java

```diff
@@ -467,13 +467,13 @@ private static byte[] readPrefix(DataInput in, ByteSequence prefixSource) throws
     return data;
   }
 
-  private static void readPrefix(DataInput in, MutableByteSequence dest, ByteSequence prefixSource)
+  private static void readPrefix(DataInput in, ArrayByteSequence dest, ByteSequence prefixSource)
       throws IOException {
     int prefixLen = WritableUtils.readVInt(in);
     int remainingLen = WritableUtils.readVInt(in);
     int len = prefixLen + remainingLen;
     if (dest.getBackingArray().length < len) {
-      dest.setArray(new byte[UnsynchronizedBuffer.nextArraySize(len)], 0, 0);
+      dest.reset(new byte[UnsynchronizedBuffer.nextArraySize(len)], 0, 0);
     }
```

Comment:
You could probably do the following here:

```
    byte[] buf = dest.getBackingArray();
    if (dest.getBackingArray().length < len) {
      buf = new byte[UnsynchronizedBuffer.nextArraySize(len)];
    } 
```

Then, everywhere below where `dest.getBackingArray` is used in the calls to `System.arraycopy`, replace them with `buf`. Then finally call `ArrayByteSequence.reset` using `buf`.

Reply:
Updated.